ci-operator/config/stackrox/kernel-packer/stackrox-kernel-packer-ddolgov-feature-repackage-cache.yaml

```diff
@@ -0,0 +1,93 @@
+build_root:
+  image_stream_tag:
+    name: release
+    namespace: openshift
+    tag: golang-1.17
+resources:
+  '*':
+    limits:
+      memory: 4Gi
+    requests:
+      cpu: 100m
+      memory: 200Mi
+tests:
+- as: kernel-crawling-periodic
+  cron: 15 */8 * * *
+  steps:
+    post:
+    - as: delete-vm
+      commands: .openshift-ci/crawler/post.sh
+      credentials:
+      - mount_path: /tmp/secret/stackrox-kernel-packer-crawl
+        name: stackrox-collector-driver-push
+        namespace: test-credentials
+      from: src
+      resources:
+        requests:
+          cpu: 4000m
+          memory: 4000Mi
+    pre:
+    - as: create-vm
+      commands: .openshift-ci/crawler/pre.sh
+      credentials:
+      - mount_path: /tmp/secret/stackrox-kernel-packer-crawl
+        name: stackrox-collector-driver-push
+        namespace: test-credentials
+      from: src
+      resources:
+        requests:
+          cpu: 4000m
+          memory: 4000Mi
+    test:
+    - as: crawling
+      commands: .openshift-ci/gcp/command.sh .openshift-ci/crawler/crawl.sh
+      credentials:
+      - mount_path: /tmp/secret/stackrox-kernel-packer-crawl
+        name: stackrox-collector-driver-push
+        namespace: test-credentials
+      from: src
+      resources:
+        requests:
+          cpu: 4000m
+          memory: 4000Mi
+      timeout: 6h0m0s
+    - as: sync
+      commands: .openshift-ci/gcp/command.sh .openshift-ci/crawler/sync.sh
+      credentials:
+      - mount_path: /tmp/secret/stackrox-kernel-packer-crawl
+        name: stackrox-collector-driver-push
+        namespace: test-credentials
```

Comment:
Not a criticism for this work, but perhaps a feature request we could make in OSCI - this credentials section is repeated for all jobs, and I wonder if it might be nice to be able to define it for a pipeline as opposed to a step within that pipeline. Now that I think about it, there's some obvious security downsides to having a global "default to having all the credentials", but we're sort of doing that anyway and the vault system lends itself to having large vaults of all the credentials, so there's still some redundancy in each step.

Like I say, not something we can change now (as far as I am aware) but perhaps something to propose as a feature.

Reply:
I wasn't actually sure if it's possible to specify credentials on the test level, to avoid repetition. But after trying make update have removed it from the top level test structure, leaving no credentials at all in the end.

About having all credentials, without planning it happened that we have only needed credentials in the vault (as the rest are located in other places now).